src/sql/workbench/services/connection/browser/connectionDialogService.ts

```diff
@@ -166,6 +166,10 @@ export class ConnectionDialogService implements IConnectionDialogService {
 				}
 				profile = result.connection;
 
+				if (params.oldProfileId && params.isEditConnection) {
+					profile.id = params.oldProfileId;
```

Comment:
Can't you just use this._inputModel.id?

(all badness of the whole storing state in the service aside - it's already there so should be usable without passing around more parameters at least)

Reply:
I'll change it to use that

Reply:
I think in the eventual pattern i'll keep this parameter passing as well.

Reply:
Alright, fair enough as a temporary thing